akka-remote/src/main/scala/akka/remote/security/provider/AES128CounterSecureRNG.scala

```diff
@@ -3,26 +3,26 @@
  */
 package akka.remote.security.provider
 
-import org.uncommons.maths.random.{ AESCounterRNG, SecureRandomSeedGenerator }
+import java.security.SecureRandom
+import java.util.concurrent.Executors
+
 import SeedSize.Seed128
 
+import scala.concurrent.ExecutionContext
+
 /**
- * INTERNAL API
- * This class is a wrapper around the 128-bit AESCounterRNG algorithm provided by http://maths.uncommons.org/
+ * This class is a wrapper around the 128-bit AESCounterBuiltinRNG AES/CTR PRNG algorithm
  * The only method used by netty ssl is engineNextBytes(bytes)
- * This RNG is good to use to prevent startup delay when you don't have Internet access to random.org
  */
 class AES128CounterSecureRNG extends java.security.SecureRandomSpi {
-  /**Singleton instance. */
-  private final val Instance: SecureRandomSeedGenerator = new SecureRandomSeedGenerator
+  private val singleThreadPool = ExecutionContext.fromExecutorService(Executors.newSingleThreadExecutor())
```

Comment:
`ThreadPoolExecutor` with 1 thread. This is enough as if the future does not complete, the generator will block and will not submit another future.